core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -281,16 +309,21 @@ class GroupMetadataManager(brokerId: Int,
 
             val responseError = group synchronized {
               if (status.error == Errors.NONE) {
-                if (!group.is(Dead)) {
+                if (!group.is(Dead) && !isTxnOffsetCommit) {
                   filteredOffsetMetadata.foreach { case (topicPartition, offsetAndMetadata) =>
                     group.completePendingOffsetWrite(topicPartition, offsetAndMetadata)
                   }
                 }
                 Errors.NONE
               } else {
                 if (!group.is(Dead)) {
+                  if (!group.hasPendingOffsetCommitsFromProducer(producerId))
+                    removeProducerGroup(producerId, group.groupId)
```

Comment:
Hmm, is this a right behavior to clear the producer group if append fails? For example if we have some previous appends for the same group that has been successful and acked, and this append failed due to an error, and the client just gives up and proceed to commit transaction, and then when the commit marker is received we should still reflect the previously acked offset in the cache right? @apurvam @hachikuji